src/pytest_inmanta_lsm/plugin.py

```diff
@@ -132,10 +135,40 @@ def remote_orchestrator_container(
         yield None
         return
 
+    orchestrator_image = inm_lsm_ctr_image.resolve(request.config)
+    latest_compose_file = pathlib.Path(__file__).parent / "resources/docker-compose.yml"
+    legacy_compose_file = pathlib.Path(__file__).parent / "resources/docker-compose-iso6.yml"
+    try:
+        compose_file = inm_lsm_ctr_compose.resolve(request.config)
+    except pytest_inmanta.test_parameter.ParameterNotSetException:
+        # The compose file is not set, we can then either use the default one, or
+        # the default legacy one (for <iso7.1).  To decide which one is the most
+        # appropriate, we parse the container image tag and extract the iso version
+        iso_major_version_match = re.fullmatch(
+            r"(?:.*)\/service-orchestrator:((\d+(?:\.\d+)*)(?:\-dev)?|dev)",
```

Comment:
Not a strict change request:

```suggestion
            r".*\/service-orchestrator:((\d+(?:\.\d+)*)(?:\-dev)?|dev)",
```
I had to look up what this meant, and I believe it's redundant so might as well keep it  simple?
Personally, I'd even go so far as to drop all of them (replacing with plain `()`) because imo it reads a lot easier.

```suggestion
            r"(?:.*)\/service-orchestrator:((\d+(\.\d+)*)(\-dev)?|dev)",
```

Reply:
Indeed, I cleaned this up.  I went for named groups, as I find it a lot easier to read when so many groups are in the regex

Reply:
I feel differently, perhaps because I'm so used to generic regexes, rather than Python's dialect. But it's a lot more clear than the non-capturing groups imo (and my objection is very subjective), so fine by me.